action.c

```diff
@@ -1484,24 +1484,27 @@ actionTryRemoveHardErrorsFromBatch(action_t *__restrict__ const pThis, wti_t *__
 {
 	actWrkrInfo_t *const wrkrInfo = &(pWti->actWrkrInfo[pThis->iActionNbr]);
 	const unsigned nMsgs = wrkrInfo->p.tx.currIParam;
-	actWrkrIParams_t oneParamSet[CONF_OMOD_NUMSTRINGS_MAXSIZE];
+	actWrkrIParams_t *oneParamSet = NULL;
 	rsRetVal ret;
 	DEFiRet;
+	CHKmalloc(oneParamSet = malloc(sizeof(actWrkrIParams_t) * pThis->iNumTpls));
 
 	*new_nMsgs = 0;
 	for(unsigned i = 0 ; i < nMsgs ; ++i) {
 		setActionResumeInRow(pWti, pThis, 0); // make sure we do not trigger OK-as-SUSPEND handling
-		memcpy(&oneParamSet, &actParam(wrkrInfo->p.tx.iparams, pThis->iNumTpls, i, 0),
+		memcpy(oneParamSet, &actParam(wrkrInfo->p.tx.iparams, pThis->iNumTpls, i, 0),
 			sizeof(actWrkrIParams_t) * pThis->iNumTpls);
 		ret = actionTryCommit(pThis, pWti, oneParamSet, 1);
 		if(ret == RS_RET_SUSPENDED) {
-			memcpy(new_iparams + *new_nMsgs, &oneParamSet,
+			memcpy(new_iparams + (*new_nMsgs * pThis->iNumTpls), oneParamSet,
```

Comment:
I tend to think this is the real bug and it stems back to the invalid not-use of the param access macro (which the new code mimics).

Reply:
i agree, now the PR os modified just to include this and one trivial initialization fix.

Reply:
thx a lot, much appreciated!